Bunch of Bandit Updates

Dedwolf opened the pull request:
## About The Pull Request

Lots of changes.

- Fixed the bug causing bandits to be assigned to Matthios regardless of Inhumen patron choice, they are intended to be allowed any choice of Inhumen patron.
- Provided a choice on spawn between the steel mask and the wildguard mask, because they're cool. I did this because I had no idea the parity merge let you swap between steel mask and steel hound mask freely, but I kept it mask swap feature in by letting you choose between the steel mask or the wildguard instead, might still remove this one.
- Gave all bandit classes steelhearted.
- Buffed sellsword with 1 additional point in Per to bring them in line with the other classes in terms of point totals.
- Pioneer gets deathbysnusnu, they're dorky but not that dorky. 
- Created subtypes of the garrote and blackbag for non-psydonite use. 
- Added the garrote and blackbag to the Brigand, Hedgeknight, Iconoclast and Sellsword as a means of silencing those they kidnap, mask covering or not.
- Provided combat oriented bandits with the blackbagger trait to utilize the garrote and blackbag.
- Provided all bandits with at least some tracking capability, either novice or apprentice if they didn't already have any.
- Buffed the hedgeknight's skills with other weapons, buffed their wrestling, gave them some knife skill, in line with the regular knight class.
- Proliferated a bunch of armor between bandit classes. 
- Added prosthetics to the bandit shop, they're expensive but much cheaper for the sawbones. 
- Added sheaths, emberwine potion, rot cure potion, firebombs, leashes and collars, lockpicks, grappling hook, climbing gear, hardtack and scrolls to the general bandit shop. 
- Added some poisons to the knave's shop. 

More to come, suggest changes in the comments.

## Testing Evidence

<img width="1848" height="1110" alt="image" src="https://github.com/user-attachments/assets/075d9e3f-2e41-4151-a59e-24999d9664ee" />

## Why It's Good For The Game

Most controversial thing here is gonna be the grappling hook and the climbing gear. I've priced them in such a way that by purchasing them you very well may be ruining your chances of getting a dragonscale necklace, or, if you have the dragonscale necklace, you still have some more gadgets worth buying, and more stuff worth stealing. The climbing gear lets you rappel down a z-level, apparently, i've never actually seen it in use. The grappling gun is stupid expensive you're straight up better stealing one or making an artificer get you one. Grappling hook is 600 points, climbing gear is 400 points.

Emberwine? Fuck yeah you know what this is for, it is priced high though because it could be used as a poison. 150 points.
Rot cure potion? With the ability for players to spawn endless mobs on mount decap and the likelihood of deadite volves roaming the rockhill bog yeah, this is something we agonizingly need, not just for bandits but for the players kidnapped or roped into their schemes. Priced high at 200 points because that's part of the sawbones' job, might reduce it.

Lockpicks? Yep, priced a bit higher than what the knave gets, the other classes don't get lockpicking skill so its mostly useful if you're running the larcenous virtue. 

Blackpowder stick? Yep, they're great for when you need to bust in somewhere quick, players can pick them up and snuff them out and keep them if they see it. I've also seen plenty of bandits blow themselves up with it. Priced at 60 points, fair bit more expensive than what the pioneer has to pay.

Heheniy commented on code/modules/antagonists/roguetown/villain/bandit.dm:
```diff
@@ -29,8 +29,6 @@
 	owner.current.playsound_local(get_turf(owner.current), 'sound/music/traitor.ogg', 60, FALSE, pressure_affected = FALSE)
 	var/mob/living/carbon/human/H = owner.current
-	if(!istype(H.patron, (/datum/patron/inhumen/matthios || /datum/patron/inhumen/graggar || /datum/patron/inhumen/zizo || /datum/patron/inhumen/baotha)))
-		H.set_patron(/datum/patron/inhumen/matthios)	//If you aren't a heretical worshiper, forces you to Matthios worship. (All bandits follow Matthios.)
```
it literally shows how it was and it wasnt like that

ProbablyCarl commented:
Hard veto on the garrote. They were just heavily buffed(garrote), to the point of being useful for actually subduing folks again, and still function as an exclusive item for Confessors (for the most part). It's part of the kit, alongside the slurbow, that makes them unique(knave as an exception now). Doubly so given as they've no miracles or anything of the sort, unlike other Psydonites. Even if they had miracles though, my argument stands.

Bandits, by design, have much higher wrestling and other stats (almost across the board) to facilitate grapple fuckery. Knave and Sawbones being an exception, I suppose. But still. This would make the garrote much worse to deal with. God forbid Iconoclast gets it.

Blackbag they don't need the trait for. Remove it. It halves the application time, and they're not trained spies / assassins, unlike Confessors. I'm not a fan of them getting this either, and suggest they stick with sacks. Of which they have _many_. Blackbags outright mute/silence a player, block sight, of course, and on top of that make it harder to resist out of grabs.

Not a fan of grappling hooks or climbing gear for them in the hoardmaster, either. Super powerful.

Additionally, Sawbones has firebombs already. You're doubling up by giving an option to everyone in the all access category. Also, for the love of god, don't give them blastpowder sticks universally ZIZUS PYST. That's not mentioning that they've a weird path, too, of 'blackmarket_diplomacy'. Why?

Emberwine is also readily available for Sawbones. No need to double up on that, too.

The rest of the PR is fine. You also don't need to apply Steelhearted to each individual class. Just give it to the parent datum, which already has the crafting trait for all of them.

Skyglazer commented:
Very much agree with Carl; bandits do not need garrotes and black-bags. Hedge knight also does not need a buff, nor do general bandits need access to things like **the grappling hook** (good god), firebombs, etc. The town already struggles enough to defend against bandits at their current power level.

Bandits are strong enough already. These buffs are too much. Perhaps these would be warranted if they were a once-in-a-while antag, like SS13 nukeops-- but these guys can show up literally every round-- they do NOT need to be this strong, nor do they need the ability to buy blastpowder sticks to blow shit up, considering one of the maps in our rotation is the already very exposed Dun World-- you don't need to be blasting holes in places to gain entry, I promise you-- especially with advanced climbing in the game. 

A gratuitous, unneeded bandit buff- this should not be added.

Dedwolf commented:
Made a few changes. 

- Moved the garrote and blackbag OUT of the general category and added it only to the classes that get the blackbagger trait, that being Brigand, Hedgeknight, Sellsword and Iconoclast. 
- Fixed some path issues from writing this at 3am. 
- Added Steelhearted and Deathbysnusnu to the bandit.dm file so those traits are now inherited by all types of bandits no matter what. 
- Increased the cost of the grappling hook and climbing gear to 800 and 600.

RadiantFlash7 commented on code/modules/jobs/job_types/roguetown/adventurer/types/antag/hedgeknight.dm:
```diff
@@ -18,11 +18,12 @@
 	)
 	subclass_skills = list(
-		/datum/skill/combat/polearms = SKILL_LEVEL_JOURNEYMAN,
+		/datum/skill/combat/polearms = SKILL_LEVEL_EXPERT,
```
I don't think the HK should be getting expert in all weapon types. They're only given swords in their shop to begin with.

Jamdrawers commented:
> Hedgie def needs a buff he's sooo pathetic next to Iconoclast LMAO
> 
> Poor little martial with less Strength and master skills than the fullcleric...

Honestly I feel like the problem there is iconoclast rather than anything else. It's a very silly role.
